File: 179.py
class Solution:
    # @param {integer[]} nums
    # @return {string}
    def largestNumber(self, nums):
        nums = sorted([str(item) for item in nums])
        nums.sort(cmp = lambda x,y: cmp(y+x,x+y))
        return ''.join(nums).lstrip('0') or '0'

File: 179.cpp
class Solution {
public:
    static bool cmp(int a, int b){
        string as = to_string(a),bs = to_string(b);
        return as + bs > bs + as;
    }
    
    string largestNumber(vector<int>& nums) {
        sort(nums.begin(), nums.end(), cmp);
        string res;
        for (auto x: nums){
            res += to_string(x);
        }
        if (res[0] == '0'){
            return "0";
        }
        return res;
    }
};
